src/be/nn.ts

```diff
@@ -0,0 +1,176 @@
+// This is adapted from work by Cédric Krier Copyright (C) 2021-2022
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
```

Comment:
I would rather not include any dependancies that change the licensing terms of the codebase.  Right now this package is covered by the MIT license and the LGPL only makes things harder for folks in the future.

Reply:
If this was a direct copy we could have a long chat, but given that outside the basics of validation your code is substantially different I would recommend removing the copyright statement all together since while the algorithm is based on the standard and the example given in the referenced code the implementation is different.